File: src/main/java/com/bermudalocket/terracotta/command/EntityCountCommand.java
package com.bermudalocket.terracotta.command;

import com.bermudalocket.terracotta.Terracotta;
import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.CommandPermission;
import dev.jorel.commandapi.arguments.Argument;
import dev.jorel.commandapi.arguments.EntityTypeArgument;
import org.bukkit.Bukkit;
import org.bukkit.entity.EntityType;

import java.util.LinkedHashMap;

public class EntityCountCommand {

    private EntityCountCommand() { }

    public static void register() {
        var argsMap = new LinkedHashMap<String, Argument>();
        argsMap.put("entity-type", new EntityTypeArgument());

        new CommandAPICommand("ec")
            .withPermission(CommandPermission.fromString("LT.admin"))
            .withArguments(argsMap)
            .executesPlayer((player, args) -> {
                var type = (EntityType) args[0];
                var r = Terracotta.getViewDistance();
                var count = Bukkit.getOnlinePlayers()
                      .stream()
                      .flatMap(p -> p.getNearbyEntities(r, r, r).stream().filter(e -> e.getType() == type))
                      .count();
                player.sendMessage("[Terracotta] Found " + count + " " + type.toString());
            })
            .register();
    }

}

File: src/main/java/com/bermudalocket/terracotta/command/EntityCountCommand.kt
package com.bermudalocket.terracotta.command

import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.EntityType
import org.bukkit.entity.Player

object EntityCountCommand: CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (sender is Player) {
            var type: EntityType? = null
            for (i in args.indices) {
                if (args[i] == "-e") {
                    try {
                        type = EntityType.valueOf(args[i + 1])
                    } catch (e: Exception) {
                        sender.sendMessage("${ChatColor.RED}Not a valid entity type: ${args[i+1]}")
                    }
                }
            }
            if (type == null) {
                sender.sendMessage("${ChatColor.RED}You must supply an EntityType, e.g. -e IRON_GOLEM")
                return true
            }
            val entityType = type
            val count = Bukkit.getWorlds()
                    .flatMap { it.livingEntities }
                    .filter { it.type == entityType }
                    .count()
            sender.sendMessage("${ChatColor.GRAY}Found $count ${type}s.")
        }
        return true
    }

}
